internal/backend/service/event_test.go

```diff
@@ -69,22 +56,11 @@ func TestEventService_PublishEvent(t *testing.T) {
 }
 
 func TestEventService_GroupFiltering(t *testing.T) {
-	// Create in-memory SQLite database for testing
-	db, err := sql.Open("libsql", ":memory:")
-	if err != nil {
-		t.Fatalf("Failed to create test database: %v", err)
-	}
-
-	defer func() {
-		if err := db.Close(); err != nil {
-			t.Error("Failed to close database", "error", err)
-		}
-	}()
-
 	group1ID := "group-1"
 	group2ID := "group-2"
 
-	service := NewEventService(db)
+	// PublishEvent doesn't use dbClient, so we can pass nil
+	service := NewEventService(nil)
```

Comment:
We should just remove the dbClient from event service so we aren't passing meaningless arguments

NOTE: This is my fault